api/v1alpha3/awsmachine_webhook.go

```diff
@@ -38,12 +38,23 @@ func (r *AWSMachine) SetupWebhookWithManager(mgr ctrl.Manager) error {
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=false,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=validation.awsmachine.infrastructure.cluster.x-k8s.io
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=true,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=default.awsmachine.infrastructure.cluster.x-k8s.io
 
 var _ webhook.Validator = &AWSMachine{}
+var _ webhook.Defaulter = &AWSMachine{}
+
+// Default implements webhook.Defaulter so a mutating webhook will be registered for the type
+func (r *AWSMachine) Default() {
+	if r.Spec.CloudInit == nil && r.Spec.ProviderID == nil {
```

Comment:
@detiber @vincepri @rudoi tagging you for 👀 and wisdom 😄 

We may need to be careful here. This is adding a new defaulting webhook, but we also have a validating webhook that disallows most changes to the spec.

I ran into a problem where I couldn't delete my AWSMachine because I changed the rules mid-flight: I created an AWSMachine that did not default to encrypted, then I changed the CAPA code so it did. When I tried to delete the AWSMachine, the new rules flipped encrypted to true (it was trying to do an update to remove the finalizer), but that change isn't allowed in ValidateUpdate.

Now, changing the rules for when encrypted defaults to true hopefully isn't a real world test, but we will have existing v1alpha2 machines that need to be deletable, so for now we only default encrypted to true if the AWSMachine doesn't yet have a provider id. Hopefully that should be sufficient, but I'd love for you to think about this too.

Reply:
Can we do defaulting only on Create?

Reply:
If we adjust the webhook verbs, yes, but usually you want the same defaults to be applied any time you touch the object

Reply:
We *could* potentially remove the 'update' verb from the default webhook annotation above, which should only allow defaulting on create, however that could potentially cause us issues down the line if we add a new field that we expect to be able to default.

Reply:
I'm almost tempted to say that if the default is true, we should make the struct a non-pointer and have a toggle to disable encryption, rather than vice-versa. Something like `insecureSkipEncryption` or similar. This way the default is always going to be true for v1alpha3.

When we backport, we probably want the vice-versa, but that's going to be a little different given that v1alpha2 doesn't have webhooks.

What do you all think?

Reply:
I've thought about this some more... the logic we currently fills in CloudInit if it's nil. That says to me "CloudInit is required and therefore can't be a pointer". Next, we want to default Encrypted if it's not set. That means we need to distinguish between unset and a real value (true or false). That says to me "Encrypted needs to be a pointer". I think if we make those changes, it solves our issues... right?

Reply:
Switched over